libs/product/src/reducers/product-entities/product-entities.reducer.spec.ts

```diff
@@ -1,30 +1,28 @@
-import { DaffProductFactory, DaffProductModificationFactory } from '@daffodil/product/testing';
+import { DaffProductFactory } from '@daffodil/product/testing';
 
-import { DaffProductLoadSuccess, DaffProductModify } from '../../actions/product.actions';
+import { DaffProductLoadSuccess } from '../../actions/product.actions';
 import { DaffProductGridLoadSuccess, DaffProductGridReset } from '../../actions/product-grid.actions';
-import { initialState, daffProductEntitiesReducer } from './product-entities.reducer';
+import { daffProductEntitiesReducer } from './product-entities.reducer';
 import { DaffBestSellersLoadSuccess } from '../../actions/best-sellers.actions';
 import { DaffProduct } from '../../models/product';
-import { DaffProductModification } from '../../models/product-modification';
+import { daffProductEntitiesAdapter } from './product-entities-reducer-adapter';
 
 describe('Product | Product Entities Reducer', () => {
 
   let productFactory: DaffProductFactory;
-  let productModificationFactory: DaffProductModificationFactory;
 
   beforeEach(() => {
     productFactory = new DaffProductFactory();
-    productModificationFactory = new DaffProductModificationFactory();
   });
 
   describe('when an unknown action is triggered', () => {
 
     it('should return the current state', () => {
       const action = {} as any;
 
-      const result = daffProductEntitiesReducer(initialState, action);
+      const result = daffProductEntitiesReducer(daffProductEntitiesAdapter<DaffProduct>().getInitialState(), action);
 
-      expect(result).toBe(initialState);
+      expect(result).toEqual(daffProductEntitiesAdapter<DaffProduct>().getInitialState());
     });
```

Comment:
It should only ever be needed once. The only other time it would be used is if an end user makes their own reducer to inject. But at that point, do we also need a const for `daffProductEntitiesAdapter<T>().upsertMany`, `daffProductEntitiesAdapter<T>().upsertOne`, etc?

Reply:
>It should only ever be needed once.

You call it 6 times in this file.

> do we also need a const for daffProductEntitiesAdapter<T>().upsertMany, daffProductEntitiesAdapter<T>().upsertOne, etc?

It certainly won't hurt anything to create local variables for those calls where you use them.

Reply:
Oh, you mean in the spec file. done